### DISCUSSION TITLE
Check that all (public) attributes have been declared as type annotations?

### DISCUSSION THREAD

**Discussion by Tomaz-Vieira** [microsoft/pyright#4065] [Q&A]
Is it possible to enforce that a class has no attributes that haven't been declared as type hint annotations in the class body? I need something stricter than `reportUninitializedInstanceVariable`, because that option will still allow attributes to be `declared` inside `__init__`.

I think this is probably equivalent to something like type-hinted `__slots__`.

**Comment by erictraut** [comment]
No, there's no rule that enforces this in pyright today. What you are suggesting is not required for type safety, so it would be more of a coding convention — something that would be more appropriate for a linter to enforce. Perhaps pylint has such a check?
